roles/network/tasks/detected_network.yml

```
@@ -260,6 +264,16 @@
     wifi_up_down: False
   when: rpi3bplus_rpi4_wifi_firmware == "24"
 
+- name: Detect "Firmware rejected country setting" in dmesg
+  shell: dmesg | grep ieee80211 | grep "Firmware rejected country setting"
+  register: FW_rejected_country
+  ignore_errors: True
+
+- name: Detect country code passed from cmdline in dmesg
+  shell: dmesg | grep ieee80211 | awk -F cfg80211.ieee80211_regdom= '{print $2}'
```

- `-m1` keeps grep from returning too many results.
- `'cfg80211\.ieee80211_regdom='` is a lot safer than `ieee80211` with grep

```suggestion
  shell: dmesg | grep -m1 'cfg80211\.ieee80211_regdom=' | awk -F cfg80211.ieee80211_regdom= '{print $2}'
```

---

Oops, I forgot one.  Pattern strengthened on left and right side too.

---

Only matters if you are going to push this, I'm comfortable with the way it is now and is tested as such. You break it you bought it.

---

Oops our awk attempts above were dependent on `cfg80211.ieee80211_regdom=XX` being on the right side of the line.  Likely a fragile assumption.  This is much less brittle code:

`dmesg | grep -om1 'cfg80211\.ieee80211_regdom=\S*' | cut -d= -f2`

ASIDE: Both `awk` and `cut` do not provide graceful error handling when pattern doesn't match.  If in future we want to force exit status (STDERR) to be non-zero (e.g. allowing Ansible to act programmatically on `cmdline_country_code.rc`) this does the same thing with cleaner error handling:

`tmp=$(dmesg | grep -om1 'cfg80211\.ieee80211_regdom=\S*') && echo $tmp | cut -d= -f2`

---

Mine is easier to read.

---

My concern is that `awk -F cfg80211.ieee80211_regdom=` generates the wrong output — it outputs the entire right-side of that `dmesg` line erroneously — when kernel parameters appear in a different order.

---

Fair enough, there is the possibility of a user tacking something on the end for whatever reason.